src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -432,17 +431,15 @@ private bool ShouldBeInFullSyncMode(Snapshot best)
 
             // Shared with fast sync
             bool notInBeaconModes = !best.IsInAnyBeaconMode;
-            bool postPivotPeerAvailable = AnyPostPivotPeerKnown(best.Peer.Block);
+            bool postPivotPeerAvailable = best.AnyPostPivotPeerKnown;
 
             // Shared with full sync archive
             bool desiredPeerKnown = AnyDesiredPeerKnown(best);
 
             // Full sync specific
-            bool hasFastSyncBeenActive = best.Header >= _pivotNumber;
+            bool hasFastSyncBeenActive = best.Header >= best.PivotNumber;
             bool notInFastSync = !best.IsInFastSync;
             bool notInStateSync = !best.IsInStateSync;
-            bool stateSyncFinished = best.State > 0
```

Comment:
no need to check for that?

Reply:
Should be covered by fast sync and state sync.